src/NuGet.Services.Metadata.Catalog.Monitoring/Status/PackageMonitoringStatusService.cs

```diff
@@ -86,17 +87,27 @@ public async Task<PackageMonitoringStatus> GetAsync(FeedPackageIdentity package,
             }
 
             // If more than one status exist for a single package, find the status with the latest timestamp.
+            // We then must delete the other statuses, so that we can later update the storage safely.
             var statusesWithTimeStamps = statuses.Where(s => s.ValidationResult != null && s.ValidationResult.CatalogEntries != null && s.ValidationResult.CatalogEntries.Any());
+            IEnumerable<PackageMonitoringStatus> orderedStatuses;
             if (statusesWithTimeStamps.Any())
             {
-                return statusesWithTimeStamps.OrderByDescending(s => s.ValidationResult.CatalogEntries.Max(c => c.CommitTimeStamp)).First();
+                orderedStatuses = statusesWithTimeStamps.OrderByDescending(s => s.ValidationResult.CatalogEntries.Max(c => c.CommitTimeStamp));
             }
             else
             {
                 // No statuses have timestamps (they all failed to process).
                 // Because they are all in a bad state, choose an arbitrary one.
-                return statuses.First();
+                orderedStatuses = statuses;
```

Comment:
Consider preferring invalid statuses over valid and pending statuses, and consider preferring pending statuses over valid statuses.